File: test/fixtures/ts-app-bulk/cart.test.js
import assert from 'node:assert/strict';
import { test } from 'node:test';
import { pricePerItem } from './cart.js';
import { bulkUnitPrice } from './bulk.js';

test('pricePerItem rounds to the nearest cent', () => {
  // 1007 / 4 = 251.75, which rounds up, not down. This is the seeded bug.
  assert.equal(pricePerItem(1007, 4), 252);
});

test('bulkUnitPrice rounds to the nearest cent too', () => {
  // Copy-pasted from cart.js; fixing only one file leaves this red.
  assert.equal(bulkUnitPrice(1007, 4), 252);
});

test('an exact division is unaffected either way', () => {
  assert.equal(pricePerItem(1000, 4), 250);
  assert.equal(bulkUnitPrice(1000, 4), 250);
});

File: test/fixtures/go-app-bulk/cart_test.go
package bulkpricing

import "testing"

func TestPricePerItemRoundsToNearestCent(t *testing.T) {
	// 1007 / 4 = 251.75, which rounds up, not down. This is the seeded bug.
	if got := PricePerItem(1007, 4); got != 252 {
		t.Errorf("PricePerItem(1007, 4) = %d, want 252", got)
	}
}

func TestBulkUnitPriceRoundsToNearestCentToo(t *testing.T) {
	// Copy-pasted from cart.go; fixing only one file leaves this red.
	if got := BulkUnitPrice(1007, 4); got != 252 {
		t.Errorf("BulkUnitPrice(1007, 4) = %d, want 252", got)
	}
}

func TestExactDivisionIsUnaffectedEitherWay(t *testing.T) {
	if got := PricePerItem(1000, 4); got != 250 {
		t.Errorf("PricePerItem(1000, 4) = %d, want 250", got)
	}
	if got := BulkUnitPrice(1000, 4); got != 250 {
		t.Errorf("BulkUnitPrice(1000, 4) = %d, want 250", got)
	}
}
